guides/deployments/proxy.md

```diff
@@ -18,27 +18,36 @@ configured the `coderd.proxy.http` and `coderd.proxy.https` settings in the
 [Helm chart](../admin/helm-charts.md). These settings correspond to the standard
 `http_proxy` and `https_proxy` environment variables, respectively.
 
-If the proxy URL does not include a scheme, Coder defaults to treating it as
-an HTTP proxy. Coder also supports proxies using HTTPS or SOCKS 5 protocols.
+If the proxy URL does not include a scheme, Coder defaults to treating it as an
+HTTP proxy. Coder also supports proxies using HTTPS or SOCKS 5 protocols. As a
+special case, Coder will always establish connections to `localhost` directly,
```

Comment:
there's two protocols involved here: the protocol of the endpoint we're trying to connect to (HTTP or HTTPS), and the protocol of the proxy (HTTP, HTTPS, SOCKS 5)

what this sentence was trying to indicate is that, when trying to connect to a `localhost` endpoint - regardless of protocol (HTTP or HTTPS) or port number of the endpoint - we connect directly and do not use the proxy

the reason for this is that, we're connecting to a remote proxy and then asking it to connect somewhere else. if we connect to the proxy and ask for localhost, then it will connect us to the host running the proxy, rather than the local machine